doc/classes/AudioEffectSampleRate.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="AudioEffectSampleRate" inherits="AudioEffect" version="4.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+	<brief_description>
+		Adds a sample rate audio effect to an Audio bus.
+	</brief_description>
+	<description>
+		Resamples the input audio signal at a new sample rate. Aliasing artifacts are produced in the resulting signal, since no interpolation is applied.
+		[b]Note:[/b] To reduce an audio signal's bit depth, use the "LoFi" mode in [AudioEffectDistortion].
+	</description>
+	<tutorials>
+		<link title="Audio buses">$DOCS_URL/tutorials/audio/audio_buses.html</link>
+	</tutorials>
+	<members>
+		<member name="mix" type="float" setter="set_mix" getter="get_mix" default="1.0">
+			Linear fade between dry/wet signals. Value can range from [param 0.0] to [param 1.0].
+		</member>
+		<member name="rate" type="float" setter="set_rate" getter="get_rate" default="11025.0">
+			Sample rate (in Hz) that input audio will be resampled to. Value must be at least [param 1.0], and should be less than [method AudioServer.get_mix_rate] to result in any audible effect.
```

Comment:
@Calinou When SampleRate's `rate` is set equal-to or greater-than than `AudioServer.mix_rate`, there is no audible effect. The sample & hold algorithm "keeps up" with the mix rate, and just passes the samples through at the same rate as they come in. It can only sample as fast as the samples come in, so even setting it ridiculously higher would have no further impact on the audio, nor the performance.

In regards to the original AudioStreams' sample rate:
AudioStreams are first resampled to `AudioServer.mix_rate` with Cubic interpolation before being processed by any effects. So, even if an AudioStream's quality was, say, 11025Hz, and SampleRate's `rate` was set to 22050Hz, the AudioStream would first be resampled/upsampled to `AudioServer.mix_rate` (likely 44100Hz or 48000Hz) using high-quality interpolation, then be processed by the SampleRate effect, which would still be able to produce aliasing artifacts as a result.

I mention [AudioServer.get_mix_rate](https://docs.godotengine.org/en/latest/classes/class_audioserver.html#class-audioserver-method-get-mix-rate) instead of [ProjectSettings.audio/driver/mix_rate](https://docs.godotengine.org/en/latest/classes/class_projectsettings.html#class-projectsettings-property-audio-driver-mix-rate) because, similar to [ProjectSettings.audio/driver/output_latency](https://docs.godotengine.org/en/latest/classes/class_projectsettings.html#class-projectsettings-property-audio-driver-output-latency), the project setting doesn't always reflect the actual value at runtime. For example, even though my project's [ProjectSettings.audio/driver/mix_rate](https://docs.godotengine.org/en/latest/classes/class_projectsettings.html#class-projectsettings-property-audio-driver-mix-rate) is set to 44100Hz, audio still ends up processing at my OS audio setting of 48000Hz, which can be confirmed by calling [AudioServer.get_mix_rate](https://docs.godotengine.org/en/latest/classes/class_audioserver.html#class-audioserver-method-get-mix-rate).

I already have this line in the `rate` property's description:
"should be less than [method AudioServer.get_mix_rate] to result in any audible effect."

Maybe in the description section I could change:
"Resamples the input audio signal at a new sample rate."

To:
"Lowers the input audio's sample rate."

 Would that make it more clear that this is meant only to degrade the audio? What do you think?

Reply:
Documentation looks good to me now :slightly_smiling_face: